src/environmentd/tests/sql.rs

```diff
@@ -1891,3 +1891,45 @@ fn test_introspection_user_permissions() -> Result<(), Box<dyn Error>> {
 
     Ok(())
 }
+
+#[test]
+fn test_idle_in_transaction_session_timeout() -> Result<(), Box<dyn Error>> {
+    let config = util::Config::default();
+    let server = util::start_server(config)?;
+
+    let mut client = server.connect(postgres::NoTls)?;
+    client.batch_execute("SET idle_in_transaction_session_timeout TO '4ms'")?;
+    client.batch_execute("BEGIN")?;
+    std::thread::sleep(Duration::from_millis(5));
```

Comment:
Probably should run the select in a retry because https://dev.materialize.com/api/rust-private/tokio/time/fn.sleep.html:

> some platforms (specifically Windows) will provide timers with a larger resolution than 1 ms

Which I think means the timeout might not have fired yet.